File: Horse-racing Duals.js
const N = +(readline());
let indexes = []
for (let i = 0; i < N; i++) {
    indexes.push(+readline())
}
indexes = indexes.map(i => +i).sort((a, b) => a - b)
let min = Infinity
for (let i = 0; i < indexes.length - 1; i++) {
    let diff = indexes[i + 1] - indexes[i]
    if (diff < min) {
        min = diff
    }
}
console.log(min)

File: Horse-racing Duals.py
n = int(input())
indexes = []
for i in range(n):
    indexes.append(int(input()))
indexes.sort()
min = indexes[1]-indexes[0]
for i in range( len(indexes)-1 ):
    diff = indexes[i+1] - indexes[i]
    if diff < min:
        min = diff
print(min)
